syntax = "proto3";

message person {
  string firstName = 1;
  string lastName = 2;
  int32 age = 3;
  gender gender = 4;
  address address = 5;
  repeated int32 lucky_numbers = 6 [packed=true];
  map<string, int32> things = 7;
  Foo foo = 8;
}

message address {
  string postcode = 1;
  string country = 2;
}

enum gender {
  nb = 0;
  male = 1;
  female = 2;
}

message Foo {
  oneof fooChoice {
    int32 foo_int = 1;
    string foo_string = 2;
    int32 foo_other_int = 3;
    int32 foo_yet_another_int = 4;
  }
}
